File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/bridge/transport/Train.java
package com.bigboxcode.designpattern.bridge.transport;

public class Train extends Transport {

    public Train(Seat seat) {
        super(seat);
    }

    @Override
    public void selectTransport() {
        // Write code to select transport
        System.out.println("Train selected for transport");

        // Select seat
        seat.selectSeat();
    }
}

File: design-patterns/ts/bridge/transport-seat/train.ts
// train.ts

import Seat from "./seat";
import Transport from "./transport";

class Train extends Transport {
    constructor(seat: Seat) {
        super(seat);
    }
    selectTransport(): void {
        console.log("Train selected for transport");
        this.seat.selectSeat();
    }
}

export default Train;